src/reporter.test.ts

```diff
@@ -638,10 +640,42 @@ describe('reporter', () => {
       ];
       const expectedResultSize = getJSONBase64Size(spans);
 
-      const actual = JSON.parse(reporter.forgeAndScrubRequestBody(spans, expectedResultSize));
+      const actual = JSON.parse(reporter.forgeAndScrubRequestBody(spans, expectedResultSize)!);
       expect(actual).toEqual(expected);
     });
 
+    test('forgeAndScrubRequestBody - response with malformed JSON should be treated as string', () => {
+      const sendTime = 1234;
+      const receivedTime = 1256;
+
+      const dummyEnd = 'dummyEnd';
+      const spans = [
+        {
+          truncated: false,
+          info: {
+            httpInfo: {
+              host: 'your.mind.com',
+              request: {
+                host: 'your.mind.com',
+                headers: { 'content-type': 'application/json' },
+                body: '{"secret":"fo',
+                sendTime,
+              },
+              response: {
+                headers: {},
+                body: 'foo',
+                statusCode: 200,
+                receivedTime,
+              },
+            },
+          },
+        },
+        { dummyEnd },
+      ];
+
+      expect(spans[0]?.info?.httpInfo.request.body).toBe('{"secret":"fo');
```

Comment:
should the "fo" not be masked in this case?

Reply:
No, because that thing is not actually JSON. If we untruncate it, we lie in the payload for the likely reason of issues to the end user.